feat(component): enhance chat input component

raphaelchristi opened the pull request:
## Description
This PR enhances the Chat Input component by simplifying the input structure and improving text handling.

---

### **Changes**
- **Improved Text Handling**:
  - Replaced `MultilineInput` with `StrInput` for more efficient text management.

- **Playground Integration**:
  - Removed input handle to streamline functionality and enhance integration with the playground.

- **Image and File Support**:
  - Maintained support for images and text files through `FileInput` using `TEXT_FILE_TYPES + IMG_FILE_TYPES`.

- **Styling and Session Management**:
  - Retained essential styling and session management options to ensure consistent user experience.

---

### **Code Comparison**

#### **Before**
```python
inputs = [
    MultilineInput(
        name="input_value",
        display_name="Text",
        value="",
        info="Message to be passed as input.",
    ),
    # ...
]
```

#### **After**
```python
inputs = [
    StrInput(
        name="input_value",
        display_name="Text",
        value="",
        info="Message to be passed as input.",
    ),
    # ...
]
```

---

### **Benefits**
- Simplified code for better readability and maintenance.
- Enhanced compatibility with the playground environment.
- Continued support for multimedia input types.
- Improved overall user experience and functionality.

erichare reviewed (commented):
What are the advantages of using a StrInput over a MultilineInput for this case?

raphaelchristi commented:
> What are the advantages of using a StrInput over a MultilineInput for this case?

The main reason for using `StrInput` instead of `MultilineInput` is that both provide the same functionality, but `StrInput` does not include a handle. This makes the Chat Input component cleaner and more streamlined.  

Additionally, as the Chat Input is a primary component, having a handle is unnecessary.
Removing the handle ensures a simpler and more intuitive design, focusing on the component's core functionality.

erichare commented on src/backend/base/langflow/components/inputs/chat.py:
```diff
@@ -13,7 +13,7 @@ class ChatInput(ChatComponent):
 
     inputs = [
-        MultilineInput(
+        StrInput(
```
@raphaelchristi Could we do this instead:

1. Keep it as a MultilineInput as before
2. But, pass `input_types=[]` so that the handle should disappear

Multiline has some nice properties when composing the text, so i think if we do this, we get the advantage of your PR (the handle being hidden for UI cleanliness) but also the multiline advantage.

raphaelchristi replied:
Thank you @erichare for this suggestion! I agree this is a much better approach.

erichare reviewed (commented):
@ogabrielluiz @raphaelchristi looking at the CI issue, it seems at least one of our tests has a component that does indeed pipe data into a chat input - is this a valid flow? If so, do we want to scrap this PR?

erichare reviewed (commented):
@raphaelchristi it looks like some unit tests are failing still. I think this is a good addition... let me know if you have a chance to take a look.

raphaelchristi commented:
> @raphaelchristi it looks like some unit tests are failing still. I think this is a good addition... let me know if you have a chance to take a look.

Hi @erichare , thanks for flagging this! I've taken a look and made the following changes:

- Fixed the type mismatch in the cycle detection test by using router.message instead of router.false_response
- Added skip conditions for OpenAI-dependent tests when no API key is configured
- Verified all tests are now passing (3 passed, 3 skipped)
The changes are in commit 297bacf with message: 'Fix type mismatch in cycle test and skip OpenAI-dependent tests

erichare commented:
> > @raphaelchristi it looks like some unit tests are failing still. I think this is a good addition... let me know if you have a chance to take a look.
> 
> Hi @erichare , thanks for flagging this! I've taken a look and made the following changes:
> 
> * Fixed the type mismatch in the cycle detection test by using router.message instead of router.false_response
> * Added skip conditions for OpenAI-dependent tests when no API key is configured
> * Verified all tests are now passing (3 passed, 3 skipped)
>   The changes are in commit [297bacf](https://github.com/langflow-ai/langflow/commit/297bacf80a4957f542e3e1500cedce2ba5f35cc1) with message: 'Fix type mismatch in cycle test and skip OpenAI-dependent tests

fantastic, thanks @raphaelchristi
